### PR TITLE
Added time format into the CloudNet-Report module

### PR DESCRIPTION
**Pull request by a6doff**
### This pull request includes:

The addition of an option (true by default) to store records into subfolders with the date.

### Changes made to the repository:

Changed only the CloudNet-Report module

### Documentation of test results:

Adds "YYYY-mm-dd" (2021-06-12 for instance) into the records, option can be disabled.
_There definitely is room for optimization especially the call to check if the date has changed every time a server stops._

### Related issues/discussions:

Absolutely none

### CODE REVIEW

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 1]
```diff
@@ -79,6 +81,8 @@ public void init() {
   public void initConfig() {
     this.getConfig().getBoolean("savingRecords", true);
     this.getConfig().getString("recordDestinationDirectory", "records");
+    this.getConfig().getBoolean("addCustomDate", true);
+    this.getConfig().getString("dateFormat", "YYYY-MM-dd");
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]: I assume this should be `yyyy-MM-dd`

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 2]
```diff
@@ -88,8 +92,15 @@ public void initConfig() {
 
   @ModuleTask(order = 126, event = ModuleLifeCycle.STARTED)
   public void initSavingRecordsDirectory() {
+    SimpleDateFormat dateFormat = new SimpleDateFormat("YYYY-MM-dd");
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]: Can't we create the format once? And why is the config option for the format not used?

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 3]
```diff
@@ -88,8 +92,15 @@ public void initConfig() {
...
+    String getdate = dateFormat.format(new Date());
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
    String date = dateFormat.format(System.currentTimeMillis());
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 4]
```diff
@@ -88,8 +92,15 @@ public void initConfig() {
 
   @ModuleTask(order = 126, event = ModuleLifeCycle.STARTED)
   public void initSavingRecordsDirectory() {
-    this.savingRecordsDirectory = this.getModuleWrapper().getDataDirectory()
-      .resolve(this.getConfig().getString("recordDestinationDirectory"));
+    SimpleDateFormat dateFormat = new SimpleDateFormat(this.getConfig().getString("dateFormat"));
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]: We should create the format once, not every time

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 5]
```diff
@@ -88,8 +92,15 @@ public void initConfig() {
...
+    String getdate = dateFormat.format(new Date());
+    if (this.getConfig().getBoolean("addCustomDate") == true) {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
    if (this.getConfig().getBoolean("addCustomDate")) {
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/listener/CloudNetReportListener.java [hunk 6]
```diff
@@ -49,6 +49,8 @@ public void handle(Event event) {
 
   @EventListener
   public void handle(CloudServicePostStopEvent event) {
+    this.reportModule.initSavingRecordsDirectory();
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/listener/CloudNetReportListener.java]: This should be in the handler of the `CloudServicePreDeleteEvent`

**Review by derklaro — CHANGES_REQUESTED** [review]

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 7]
```diff
@@ -79,17 +81,28 @@ public void init() {
...
+    }
   }
 
+
   @ModuleTask(order = 126, event = ModuleLifeCycle.STARTED)
   public void initSavingRecordsDirectory() {
-    this.savingRecordsDirectory = this.getModuleWrapper().getDataDirectory()
-      .resolve(this.getConfig().getString("recordDestinationDirectory"));
+    if (this.getConfig().getBoolean("addCustomDate")) {
+      String date = dateFormat.format(System.currentTimeMillis());
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
      String date = this.dateFormat.format(System.currentTimeMillis());
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 8]
```diff
@@ -65,6 +66,7 @@
   private final IFormatter logFormatter = new DefaultLogFormatter();
   private Path savingRecordsDirectory;
   private volatile Class<? extends Event> eventClass;
+  private SimpleDateFormat dateFormat;
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
  private DateFormat dateFormat;
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/listener/CloudNetReportListener.java [hunk 9]
```diff
@@ -49,6 +49,7 @@ public void handle(Event event) {
 
   @EventListener
   public void handle(CloudServicePostStopEvent event) {
+
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/listener/CloudNetReportListener.java]:
```suggestion
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 10]
```diff
@@ -79,17 +81,28 @@ public void init() {
...
-
     this.saveConfig();
+    if (this.getConfig().getBoolean("addCustomDate")) {
+      dateFormat = new SimpleDateFormat(this.getConfig().getString("dateFormat"));
+    }
   }
 
+
   @ModuleTask(order = 126, event = ModuleLifeCycle.STARTED)
   public void initSavingRecordsDirectory() {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]: This should no longer be a module task, as we do init the directory inside the `CloudServicePreDeleteEvent`:
```suggestion
  public void initSavingRecordsDirectory() {
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 11]
```diff
@@ -79,17 +81,28 @@ public void init() {
...
+    this.getConfig().getBoolean("addCustomDate", true);
+    this.getConfig().getString("dateFormat", "yyyy-MM-dd");
     this.getConfig().get("pasteServerType", PasteServerType.class, PasteServerType.HASTE);
     this.getConfig().getString("pasteServerUrl", "https://just-paste.it");
     this.getConfig().getLong("serviceLifetimeLogPrint", 5000L);
-
     this.saveConfig();
+    if (this.getConfig().getBoolean("addCustomDate")) {
+      dateFormat = new SimpleDateFormat(this.getConfig().getString("dateFormat"));
+    }
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
    String dateFormat = this.getConfig().getString("dateFormat");
    if (dateFormat != null && this.getConfig().getBoolean("addCustomDate")) {
      this.dateFormat = new SimpleDateFormat(dateFormat);
    }
```

File: cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java [hunk 12]
```diff
@@ -79,17 +81,28 @@ public void init() {
...
   }
 
+
   @ModuleTask(order = 126, event = ModuleLifeCycle.STARTED)
   public void initSavingRecordsDirectory() {
-    this.savingRecordsDirectory = this.getModuleWrapper().getDataDirectory()
-      .resolve(this.getConfig().getString("recordDestinationDirectory"));
+    if (this.getConfig().getBoolean("addCustomDate")) {
```
Comment (derklaro) [inline cloudnet-modules/cloudnet-report/src/main/java/de/dytanic/cloudnet/ext/report/CloudNetReportModule.java]:
```suggestion
    if (this.dateFormat != null && this.getConfig().getBoolean("addCustomDate")) {
```

**Review by derklaro — CHANGES_REQUESTED** [review]

**Comment by a6doff** [comment]
@derklaro I merged without thinking that much about it but wouldn't `SimpleDateFormat` be more appropriate?
So people would also be able to add hours and minutes?

**Comment by derklaro** [comment]
> @derklaro I merged without thinking that much about it but wouldn't `SimpleDateFormat` be more appropriate?
> So people would also be able to add hours and minutes?

That's not how java works. If you create the date format using the `SimpleDateFormat` it will work as expected.

**Comment by a6doff** [comment]
> > @derklaro I merged without thinking that much about it but wouldn't `SimpleDateFormat` be more appropriate?
> > So people would also be able to add hours and minutes?
> 
> That's not how java works. If you create the date format using the `SimpleDateFormat` it will work as expected.

Yes, it was working as approprite, that's why I'm wondering why you [commited](https://github.com/CloudNetService/CloudNet-v3/pull/491/commits/f0ee1e5d99514e26d1ae086fbe8f21431498713b) to change it to `DateFormat` instead

**Comment by derklaro** [comment]
> > > @derklaro I merged without thinking that much about it but wouldn't `SimpleDateFormat` be more appropriate?
> > > So people would also be able to add hours and minutes?
> > 
> > That's not how java works. If you create the date format using the `SimpleDateFormat` it will work as expected.
> 
> Yes, it was working as approprite, that's why I'm wondering why you [commited](https://github.com/CloudNetService/CloudNet-v3/pull/491/commits/f0ee1e5d99514e26d1ae086fbe8f21431498713b) to change it to `DateFormat` instead

That's for abstraction reasons.

**Review by derklaro — APPROVED** [review]

**Comment by a6doff** [comment]
(oops, didn't mean to do that)

**Comment by a6doff** [comment]
BRO I pushed on the wrong shit

**Comment by derklaro** [comment]
Thank you!

**Pull request merged**